src/ManifoldsBase.jl

```diff
@@ -565,8 +565,19 @@ Saves the result in `q`.
 """
 function mid_point!(M::Manifold, q, p1, p2)
     X = log(M, p1, p2)
-    X /= 2
-    return exp!(M, q, p1, X)
+    return exp!(M, q, p1, X / 2)
+end
+
+@static if VERSION <= v"1.1"
```

Comment:
Why does this not work? It looks like exp/log should work anyways and despite that we only divide by two? Maybe `fill(X / 2)` could just be `X ./ 2`?